utils/kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/kafka/source/delayed/DelayedKafkaSourceFactory.scala

```diff
@@ -20,12 +20,30 @@ object DelayedKafkaSourceFactory {
   final val TimestampFieldParamName = "timestampField"
 
   // TODO: consider changing to lazy parameter and add the same parameter also in "not delayed" kafka sources
-  final val TimestampFieldParameter = Parameter
+  final val fallbackTimestampFieldParameter = Parameter
     .optional(TimestampFieldParamName, Typed[String])
     .copy(
       editor = Some(DualParameterEditor(simpleEditor = StringParameterEditor, defaultMode = DualEditorMode.RAW))
     )
 
+  def timestampFieldParameter(typingResult: Option[TypingResult]): Parameter = {
```

Comment:
1. Let's rename `typingResult` to sth like `kafkaRecordValueType`
2. Let's leave a TODO that it is not a final solution and in case of further development (e.g. nested fields adding etc.) we should rather focus on the final solution allowing to define spel expression.

Reply:
done.